### PR TITLE
add Sync Command for InventoryTrigger

### PR DESCRIPTION
**Pull request by Cupelt**
# Commands
```
/trg inventroy[i] <name> sync
```

### CODE REVIEW

**Comment by wysohn** [comment]
@Cupelt 
Looking good!

Actually, do me a favor to rebase this to https://github.com/TriggerReactor/TriggerReactor/tree/development_refactor/IOC branch?

Or I will have to re-do the same code again :p

**Comment by Cupelt** [comment]
Oh, I see

File: core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/location/LocationBasedTriggerManager.java [hunk 1]
```diff
@@ -389,7 +389,7 @@ public void showTriggerInfo(ICommandSender sender, ILocation clicked) {
         sender.sendMessage("- - - - - - - - - - - - - -");
         sender.sendMessage("Trigger: " + getTriggerTypeName());
         sender.sendMessage("Block Type: " + world.getBlock(clicked).getTypeName());
-        sender.sendMessage("Location: " + clicked);
+        sender.sendMessage("Location: " + clicked.toSimpleLocation());
```
Comment (Cupelt) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/location/LocationBasedTriggerManager.java:392]: i didnt touch this code, but its changed during rebase 
should i fix it?

**Comment by wysohn** [comment]
@Cupelt 
Oh opps didn't mean to close it

Probably due to conflict or something

**Comment by wysohn** [comment]
@Cupelt 
I think it might be cleaner to start from a fresh fork starting with the new `development` changes

you can probably use `cherry-pick` to move the last two commits that you have made. 

Let me know if you need help with that

**Comment by wysohn** [comment]
@Cupelt 
Sorry, I actually realized something crucial

Why do we allow inventory trigger to be running sync?

Inventory Trigger can possibly have an infinite loop by its design (in the `trigger=="init"` part), so we would need some careful planning on this

**Comment by Cupelt** [comment]
@wysohn if inventoryTrigger can use sync, 
you can pickup item by `event.setcancelled(false)` in runnning InventoryTrigger

i think that it increases user's degrees of freedom

**Comment by wysohn** [comment]
> @wysohn if inventoryTrigger can use sync, you can pickup item by `event.setcancelled(false)` in runnning InventoryTrigger
> 
> i think that it increases user's degrees of freedom

Got it,

if that's the case, we may need something other than just a sync

Well, what it does will be making it run in the server thread, which is same as sync, but it should only be working when the 'trigger' variable is 'click'

I believe simply changing 'sync' option in TriggerInfo will apply it to any 'trigger' variable

[here](https://github.com/TriggerReactor/TriggerReactor/blob/eedbcd0a0d98b278d9b7e31cf0c3049bae9d7661/core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTriggerManager.java#L154) for example, it simply uses `activate` method, which will automatically load sync config from the TriggerInfo, so simply changing `sync` in TriggerInfo can cause problem: for example, we allow infinite loop when 'trigger' variable is 'open.' So if that's the case, the infinite loop will run in the server thread, eventually stopping the server completely

**Comment by wysohn** [comment]
> @wysohn if inventoryTrigger can use sync, you can pickup item by `event.setcancelled(false)` in runnning InventoryTrigger
> 
> i think that it increases user's degrees of freedom

Or as you mentioned, it's something to do with event.setCancelled(true), so may be this is a better place to deal with: https://github.com/TriggerReactor/TriggerReactor/blob/eedbcd0a0d98b278d9b7e31cf0c3049bae9d7661/core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTriggerManager.java#L173

I will leave it up to you! But just let me know how you would fix the problem :)

**Comment by Cupelt** [comment]
I understood

**Pull request closed without merging**